vapi/appliance/networking/proxy.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0.
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networking
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/vmware/govmomi/vapi/rest"
+)
+
+const applianceProxyConfigPath = "/appliance/networking/proxy"
+const applianceNoProxyConfigPath = "/appliance/networking/noproxy"
+
+// Manager provides convenience methods to configure appliance proxy.
+type Manager struct {
+	*rest.Client
+}
+
+// NewManager creates a new Manager with the given client
+func NewManager(client *rest.Client) *Manager {
+	return &Manager{
+		Client: client,
+	}
+}
+
+// Proxy represents configuration for specific proxy - ftp, http, https.
+type Proxy struct {
+	Server   string `json:"server,omitempty"`
+	Port     int    `json:"port,omitempty"`
+	Username string `json:"username,omitempty"`
+	Password string `json:"password,omitempty"`
+	Enabled  bool   `json:"enabled,omitempty"`
+}
+
+// ProxyConfig represents configuration for vcenter proxy.
+type ProxyConfig struct {
```

Comment:
Wondering if we should name this `ProxyList` to match [the API](https://developer.vmware.com/docs/vsphere-automation/v7.0U2-deprecated/appliance/rest/appliance/networking/proxy/get/)
While this structure isn't a list itself, it does capture the data returned by the `ProxyList` API.

And same with the method below.  While the structure differs from what the API returns, I don't imagine new protocols being added to the list. And if they are, we can just add to this structure.

Reply:
Agree. Renamed the `struct` and the method